File: src/main/java/no/nav/syfo/util/PeriodeMapper.java
package no.nav.syfo.util;

import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import lombok.extern.slf4j.Slf4j;
import no.nav.syfo.domain.Periode;

import java.io.IOException;

@Slf4j
public class PeriodeMapper {

    private static ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());

    public static Periode jsonTilPeriode(final String json) {
        final String message = "Mapping av periode-json feiler";
        try {
            final Periode periode = objectMapper.readValue(json, Periode.class);
            if (periode.getTom() == null || periode.getFom() == null
                    || periode.getFom().isAfter(periode.getTom())) {
                log.error(message);
                throw new IllegalArgumentException(message);
            }
            return periode;
        } catch (JsonParseException | JsonMappingException exception) {
            log.error(message);
            throw new IllegalArgumentException(message, exception);
        } catch (IOException iOException) {
            log.error(message);
            throw new RuntimeException(message, iOException);
        }
    }
}

File: src/main/kotlin/no/nav/syfo/util/PeriodeMapper.kt
package no.nav.syfo.util

import com.fasterxml.jackson.core.JsonParseException
import com.fasterxml.jackson.databind.JsonMappingException
import no.nav.syfo.domain.Periode
import no.nav.syfo.logger
import no.nav.syfo.objectMapper
import java.io.IOException

object PeriodeMapper {
    private val log = logger()

    fun jsonTilPeriode(json: String?): Periode {
        val message = "Mapping av periode-json feiler"
        try {
            val periode = objectMapper.readValue(json, Periode::class.java)
            if (periode.fom.isAfter(periode.tom)) {
                log.error(message)
                throw IllegalArgumentException(message)
            }
            return periode
        } catch (exception: JsonParseException) {
            log.error(message)
            throw IllegalArgumentException(message, exception)
        } catch (exception: JsonMappingException) {
            log.error(message)
            throw IllegalArgumentException(message, exception)
        } catch (iOException: IOException) {
            log.error(message)
            throw RuntimeException(message, iOException)
        }
    }
}
